source/opt/optimizer.cpp

```diff
@@ -216,6 +216,12 @@ Optimizer& Optimizer::RegisterSizePasses() {
       .RegisterPass(CreateAggressiveDCEPass());
 }
 
+Optimizer& Optimizer::RegisterWebGPUPasses() {
+  return RegisterPass(CreateEliminateDeadFunctionsPass())
+      .RegisterPass(CreateAggressiveDCEPass())
+      .RegisterPass(CreateEliminateDeadConstantPass());
```

Comment:
If you really just want removal of dead code, and not something that will look to simplify obvious stuff, you should do AggressiveDCEPass and DeadBranchElimination.  The other two that you have are redundant when you have ADCE.

Reply:
Done